src/Home/AuthorisedHome.jsx

```diff
@@ -53,16 +55,19 @@ class AuthorisedHome extends React.Component {
           <Redirect to="/" noThrow />
         ) : (
           <div>
-            <UserInfo
-              userProfile={this.state.user_arr}
-              logOutHandler={this.handleLogOutClick}
-            />
+            <UserProvider value={this.state.user_arr}>
```

Comment:
Doesn't make sense to put user_arr in state when the using context. Lift the context one component higher and pass it directly from localstorage.